doc/network_scenarios.md

```diff
@@ -0,0 +1,418 @@
+</p>
+
+Nevertheless, during the first stage we could need network so we are able to configure
+it for the running system via linuxrc deciding then if such configuration should be 
+copied from the inst-sys (this decision is made based on `keep_install_network` value).
+
+Let's go with some scenarios and examples...
+
+Scenarios
+=========
+
+First of all we are going to explain what happens during the **first stage**, and then what
+happens during the **second stage** if not avoided. You can also check the current documentation
+https://github.com/yast/yast-network/blob/master/doc/autoinstallation.md#introduction for AutoYast.
+
+
+For the examples below consider this linuxrc options if not mentioned different ones:
+
+**LinuxRC options:** autoyast=http://..sle12_minimal.xml ifcfg=eth0=dhcp
+
+First Stage
+-----------
+
+The network configuration for the first stage currently defined in the control 
+file takes part in these clients (**inst_autoinit**, **inst_autosetup** and 
+**inst_finish**).
+
+- **inst_autoinit:** Autoinit will call iSCSI or FCOE clients if they are 
+  enabled in Linuxrc and will try to fetch and process the profile.
+
+- **inst_autosetup:** This client checks the profile and writes the network 
+  configuration files only if there is a network section defined 'and' (the
+  setup_before_proposal flag is true 'or' if there is a semi-automatic section)
+
+- **inst_finish:** At the end it will call **save_network** client which copies
+  udev rules and ifcfg files from the running system and is also responsible 
+  for writing several proposals like virtualization, dns and network service.
+
+Enough theory by now, it's time to the examples:
+
+1. **Autoinstallation without network section. (minimal configuration)**
+
+  Just consider the profile below as the used for this scenario.
+
+  ```xml
+<profile xmlns="http://www.suse.com/1.0/yast2ns" xmlns:config="http://www.suse.com/1.0/configns">
+  <general>
+    <mode>
+      <confirm config:type="boolean">false</confirm>
+    </mode>
+  </general>
+  <software>
+    <install_recommended config:type="boolean">true</install_recommended>
+    <patterns config:type="list">
+      <pattern>Minimal</pattern>
+      <pattern>base</pattern>
+    </patterns>
+  </software>
+</profile>
+```
+
+  **Results:**
+
+  With this configuration autoinit won't write anything because there is no networking section,
+  but as we have linuxrc network configuration the ifcfg-file exists in the running system.
+
+  ```xml
+# /etc/sysconfig/network/ifcfg-eth0 
+BOOTPROTO='dhcp'
+STARTMODE='auto'
+DHCLIENT_SET_HOSTNAME='yes' ## THIS IS NOT PRESENT IN SP3 or SP2 versions with last yast2-network package
+```
+
+  Therefore, when `save_network` is called by `inst_finish` it will copy the 
+  udev rules and sysconfig network configuration files because by default `keep_net_config?` is
+  considered as **true**.
+
+  And about DNS, as no network section is provided then it will write the proposed by 
```

Comment:
If you're using 'commas', I would go for *And about DNS, as no network section is provided, it will..."